regress/enc-dec-file.c

```diff
@@ -0,0 +1,61 @@
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <pwd.h>
+#include "includes.h"
+#include "iron-common.h"
+#include "iron-gpg.h"
+
+int
+main(int argc, char **argv)
+{
+	if (argc < 2) {
+		fprintf(stderr, "\nUsage: %s <infile>\n\n", argv[0]);
+		return -1;
+	}
+
+	if (check_iron_keys() != 0) {
+		fprintf(stderr, "Unable to find or create the necessary keys for current user's login\n\n");
+		return -2;
+	}
+	
+	//  Create a <infile>.iron file, so that write_gpg_encrypted_file will generate a different output file.
+	char tmp_fname[PATH_MAX + 1];
+	sprintf(tmp_fname, "%s%s", argv[1], IRON_SECURE_FILE_SUFFIX);
```

Comment:
Got sloppy - it was only a test file. But added a check for length of argv[1] at start of execution.